src/LogFileReader.cpp

```diff
@@ -67,19 +71,38 @@ bool LogFileReader::readAll(const QString& fileName)
             emit batteryDataReceived(batteryData);
         }
     }
-
     return true;
 }
 
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
-    // TODO implement this first
+    QStringList lines = line.split(COMMA);
+    QTime Time = QTime::fromString(lines[0], TIMEFORMAT);
+    bool ok;
+    if (Time.isValid())
+    {
+        batteryData.time = Time;
+    }
+    else
+    {
+        return false;
+    }
+
+    if (lines.count() !=    3)
+    {
+        return false;
+    }
```

Comment:
Doing the error checking together will make for shorter and easier to read code. So you can just use
```c++
if (!Time.isValid() || lines.count() != 3)
{
    return false;
}
```